R/functions.R

```diff
@@ -49,26 +61,72 @@ collapse_pm_query <- function(df) {
   entry <- dplyr::group_by(df, pmid, doi, title, abstract, year, journal, entity_name)
   entry <- summarise(entry, authors = toString(fullname))
   if (nchar(entry$authors) > 500) {
-    entry$authors <- str_trunc(entry$authors, width = 500)
+    entry$authors <- stringr::str_trunc(entry$authors, width = 500)
   }
   entry
 }
-#' Wrapper for Pubmed Query
+#' Wrapper for Pubmed Query using Pubmed IDs
 #' This function wraps several functions:
 #' - query Pubmed
 #' - create an entity name from first author, journal, year and Id
 #' - abbreviates the author names by first initial, last name
 #' - creates one row per PubmedId
 #' @param vec A character vector of PudmedIds.
+#' @param sleep_time Amount of time, in seconds, to sleep between queries.
+#' Without APIKey, Pubmed limits rate of queries to 3 per second. Default of
+#' 0.35 seconds.
 #' @export
-query_list_pmids <- function(vec) {
-  vec <- purrr::map(vec, ~ query_pm(.))
+query_list_pmids <- function(vec, sleep_time = 0.35) {
+  vec <- purrr::map(vec, ~ query_pm(., sleep_time = sleep_time))
   vec <- purrr::map(vec, ~ make_entity_name(.))
   vec <- purrr::map(vec, ~ collapse_names(.))
   vec <- purrr::map(vec, ~ collapse_pm_query(.))
   df <- data.frame(Reduce(rbind, vec))
   df
 }
+#' Wrapper for Pubmed Query using general queries
+#' This function wraps several functions:
+#' - query Pubmed
+#' - create an entity name from first author, journal, year and Id
+#' - abbreviates the author names by first initial, last name
+#' - creates one row per PubmedId
+#' - query without associated PubmedIds will be left out of returned data
+#' - query added as an additional column
```

Comment:
```suggestion
#' - query added as an additional column
```
What does this mean?

Reply:
There's an extra column in the dataframe that has the name 'query' and the string that was used as the Pubmed query.

For example, if I query using `c('foo', 'bar')`, I'll get back something like this:
| pmid  | journal | more columns | query |
| ------------- | ------------- | --- | --- |
| 12345678  | neat journal  | ... | foo |
| 88382848 | lame journal | ... | foo |
| 21324332  | cool journa  | ... | bar |

This helps with knowing what query returned a journal. In the AD case, they are using portions of the grant number as a query. I can map the full grant number to this table using the query column.